docs/components.rst

```diff
@@ -2,9 +2,128 @@
+order to associate them to the actual path permissions.
+If the path receives a ``read`` permission, every `Magpie`_ permissions fitting the ``read`` category will be enabled.
+
+If a `Magpie`_ permissions of type ``read`` is added, the path will be updated to have ``read`` permissions. This
+update on the file system will trigger a synchronization with `Magpie`_, to add all other ``read`` type permissions on
+`Magpie`_.
+
+Note that permissions are only added to `Magpie`_ if necessary. For example, if a file needs to allow a ``read``
+permission on `Magpie`_, but that permission already resolves to ``allow`` because of a recursive permission on a parent
+resource, no permission will be added. The permission is already resolving to the required permission and avoiding to
+add unnecessary permissions will simplify permission solving.
+
+The permissions applied on the files and folders are only applied for a user, and no permissions are enabled on the
+group or for other users. The reason for this is that workspaces are separated by user and we do not use a group
+concept on the file system for now. This means that if a permission is applied to a group in `Magpie`_, `Cowbird`
+will detect the permission change but will not do anything, since the group on the file system does not correspond to
+the groups found on `Magpie`_. Also, it would not make sense to update the file associated to a resource for all the
+users of a group, since the file is supposed to be associated to a single user anyway.
+
+Note that even if group permission changes on `Magpie`_ are not handled by `Cowbird`, a group permission could still
+have an impact on permission resolution. For example, if a shapefile needs to allow a ``read`` permission on `Magpie`_,
+but that permission already resolves to ``allow`` because of a group permission, no permission will be added.
+
+.. _geoserver_file_layer_permissions:
+
+File/Layer permissions
+######################
+
+Also, file events will only be processed in the case of the ``.shp`` file, since it is considered to be the main
+component of a shapefile. The other extensions associated with a shapefile will not be processed if they trigger an
+event, and will only be updated in the case of a change on the ``.shp`` file.
+
+Note that in the case where a user has all the ``read`` permissions on `Magpie`_ for example, and a single one of
+them is deleted, `Cowbird` will not change the file permissions since other ``read`` permissions will still be found on
+`Magpie`_. This means that a synchronization will not be triggered and `Magpie`_ permissions will stay the same, meaning
+all the ``read`` permissions activated except for the one removed.
+If eventually a change is applied to the file (ex.: changing the permissions from ``r--`` to ``rw-``),
+it would trigger a synchronization, and the one `Magpie`_ permission that was removed earlier would be reenabled,
+because of the ``read`` permission found on the file.
```

Comment:
Yes, what you mention is correct.
I would say the best method is to disable all readable permissions on Magpie and apply is the change all in one go. It's better than doing the changes one at a time, which would also work, but with the limitation that is a write is enabled during doing this, it would reset everything.

I will add a sentence to describe the recommandation of applying the change on all readable permissions in one go to avoid unwanted results.

Reply:
I would add that partial usage of readable or writable permissions on Magpie is not recommended since the Magpie permissions that are not yet enabled might be automatically enabled eventually if an update on the file is done, triggering a synchronization with Magpie.